src/Psl/Result/with_stats.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Result;
+
+use Psl\Fun;
+use Psl\Ref;
+use Psl\Vec;
+
+/**
+ * @template R
+ *
+ * @param callable(): iterable<array-key, ResultInterface<R>> $action
+ *
+ * @return array{0: list<ResultInterface<R>>, 1: Stats}
+ */
+function with_stats(callable $action): array
```

Comment:
alternatively we could just pass the list of results here. It makes it a bit easier to combine with e.g. the `pipe()`function.

WDYT?

Reply:
as you see fit, I'm okay with either.

Reply:
Went with the iterable solution, since it is easier to compose.
You can even compose it into something that works like the first version if you like.